cms/views/cms_views.py

```diff
@@ -318,3 +322,45 @@ def search(self):
     @notfound_view_config(renderer='cms:templates/404.pt')
     def notfound(request):
         return {}
+
+    @view_config(route_name='login')
+    def login(self):
+        hubclient = self.request.registry.hubclient
+        response = HTTPFound()
+
+        # redeem ticket to get user data
+        ticket = self.request.GET.get('ticket', None)
+        if ticket:
+            try:
+                user = hubclient.get_user(
+                    ticket, self.request.route_url('redirect_to_login'))
+                self.request.session[USER_DATA_SESSION_KEY] = user.data
+                user_id = user.get('uuid')
+                headers = remember(self.request, user_id)
+                response.headerlist.extend(headers)
+
+            except HubClientException:
+                # TODO: what to do when ticket is invalid?
```

Comment:
probably show a retry link? With SSO I found that doing a logout then asking the user to login again as a good work around for invalid tickets.